cabal-install/Distribution/Client/Dependency/Modular/Builder.hs

```diff
@@ -125,8 +125,13 @@ build = ana go
     go    (BS { index = _  , next = OneGoal (OpenGoal (Simple (Pkg _ _          ) _) _ ) }) =
       error "Distribution.Client.Dependency.Modular.Builder: build.go called with Pkg goal"
     go bs@(BS { index = idx, next = OneGoal (OpenGoal (Simple (Dep qpn@(Q _ pn) _) _) gr) }) =
+      -- If the package does not exist in the index, we construct an emty PChoiceF node for it
+      -- After all, we have no choices here. Alternatively, we could immediately construct
+      -- a Fail node here, but that would complicate the construction of conflict sets.
+      -- We will probably want to give this case special treatment when generating error
+      -- messages though.
       case M.lookup pn idx of
-        Nothing  -> FailF (toConflictSet (Goal (P qpn) gr)) (BuildFailureNotInIndex pn)
+        Nothing  -> PChoiceF qpn gr (P.fromList [])
```

Comment:
I thought that `initial` represents the `_|_` node, and `_|_` should have a conflict set containing the current variable.  Say we used an explicit `_|_` node, and it happened to be first in the list of children.  If `_|_` didn't contain the current variable, the solver would backjump after reaching the failure and would skip the siblings, even though those siblings might lead to solutions.

Since we're not using an explicit failure node, adding the current variable to `initial` wouldn't actually affect the behavior.  I just thought it might be more consistent with representing the `_|_` node.